xbmc/network/Network.cpp

```diff
@@ -206,39 +258,86 @@ bool CNetwork::IsLocalHost(const std::string& hostname)
 
 CNetworkInterface* CNetwork::GetFirstConnectedInterface()
 {
-   std::vector<CNetworkInterface*>& ifaces = GetInterfaceList();
-   std::vector<CNetworkInterface*>::const_iterator iter = ifaces.begin();
-   while (iter != ifaces.end())
-   {
-      CNetworkInterface* iface = *iter;
-      if (iface && iface->IsConnected())
-         return iface;
-      ++iter;
-   }
+   iter_for_each(GetInterfaceList(), CNetworkInterface, iface)
+     if (iface && iface->IsConnected())
+       return iface;
 
    return NULL;
 }
 
+bool CNetwork::AddrMatch(const std::string &addr, const std::string &match_ip, const std::string &match_mask)
+{
+  if (ConvIPv4(addr) && ConvIPv4(match_ip) && ConvIPv4(match_mask))
+  {
+    unsigned long address = ntohl(inet_addr(addr.c_str()));
+    unsigned long subnet = ntohl(inet_addr(match_mask.c_str()));
+    unsigned long local = ntohl(inet_addr(match_ip.c_str()));
+    return ((address & subnet) == (local & subnet));
+  }
+  else
+  {
+    struct sockaddr_in6 sa_trymatch;
+    struct sockaddr_in6 sa_iface;
+    struct sockaddr_in6 sa_mask;
+    if (!ConvIPv6(addr, &sa_trymatch) || !ConvIPv6(match_ip, &sa_iface)
+     || !ConvIPv6(match_mask, &sa_mask))
+      return false;
+
+    unsigned int m;
+    for (m = 0; m < 16;
+         ((sa_iface.sin6_addr.s6_addr[m] & sa_mask.sin6_addr.s6_addr[m])
+                                    == sa_trymatch.sin6_addr.s6_addr[m]) ? m++ : m = -1);
+    if (m < (unsigned int)~0)
```

Comment:
yes this is topic I wanted to discuss - normally I would do full comments like I did here for instance https://github.com/xbmc/xbmc/commit/31e87e2e6dd21c9d8ea5c7412e4401a114392743 .

it only doesn't seem like a widely used approach and someone can even thing of it as 'offensive'. if it's fine you get it.